app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,9 @@
+
+class FavoritesController <  ApplicationController
+  def create
+    pet = Pet.find(params[:pet_id])
+    favorites.add_pet(pet)
```

Comment:
YES this is so nice. great job creating methods like add_pet in the model so that they can be used in the controller. Very MVC